starlette/applications.py

```diff
@@ -176,7 +184,11 @@ def route(
         methods: typing.List[str] = None,
         name: str = None,
         include_in_schema: bool = True,
-    ) -> typing.Callable:
+    ) -> typing.Callable:  # pragma: nocover
+        """
+        We no longer document this decorator style API, and its usage is discouraged.
+        """
```

Comment:
I guess it would make sense for each of these docstrings to show the alternate approach in each case.

```python
routes = [
    Route(path, endpoint=..., ...),
    ...
]

app = Starlette(routes=routes)
```